cmd/ndm_daemonset/probe/eventhandler.go

```diff
@@ -58,7 +58,7 @@ func (pe *ProbeEvent) addBlockDeviceEvent(msg controller.EventMessage) {
 	// iterate through each block device and perform the add/update operation
 	for _, device := range msg.Devices {
 		klog.Infof("Processing details for %s", device.DevPath)
-		pe.Controller.FillBlockDeviceDetails(device)
+		pe.Controller.FillBlockDeviceDetails(device, msg.AllowedProbes...)
```

Comment:
Same change w.r.t variable name can be made here. RequestedProbes instead of AllowedProbes.